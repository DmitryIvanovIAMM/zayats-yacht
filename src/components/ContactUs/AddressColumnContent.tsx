import React from 'react';

export default function AddressColumnContent() {
  return (
    <div style={{ fontSize: '16px', lineHeight: '150%' }}>
      <br />
      Allied Yacht Transport, LLC
      <br />
      12555 Orange Drive, Suite 107
      <br />
      Fort Lauderdale, FL, 33330, USA
    </div>
  );
}
