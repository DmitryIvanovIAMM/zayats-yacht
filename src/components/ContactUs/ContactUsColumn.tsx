import React, { FC } from 'react';
import { Typography } from '@mui/material';
import Box from '@mui/material/Box';
import LocationOnIcon from '@mui/icons-material/LocationOn';

const contactUsContainerSx = {
  display: 'flex',
  justifyContent: 'start',
  justifyItems: 'start',
  flexDirection: 'row',
  alignItems: {
    xs: 'center',
    sm: 'start'
  }
};
const iconContainerStyle = {
  background: '#fff',
  boxShadow: '0px 2px 20px rgba(51, 101, 167, 0.18)',
  borderRadius: '50%',
  minWidth: '67px',
  maxWidth: '67px',
  height: '67px',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  marginRight: '20px'
};
const titleContainerSx = {
  textTransform: 'uppercase',
  fontWeight: '550',
  fontSize: '18px',
  lineHeight: '22px',
  alignItems: 'start',
  verticalAlign: 'top'
};

export interface ContactUsColumnProps {
  title: string;
  icon?: any;
  children: React.ReactNode;
}

const ContactUsColumn: FC<ContactUsColumnProps> = ({
  children,
  title,
  icon: Icon = LocationOnIcon
}) => {
  return (
    <Box sx={contactUsContainerSx}>
      <div style={iconContainerStyle}>
        <Icon fontSize="large" />
      </div>
      <div style={{ alignItems: 'start' }}>
        <Typography variant="h6" component="h6" sx={titleContainerSx}>
          <br />
          {title}
        </Typography>
        {children}
      </div>
    </Box>
  );
};

export default ContactUsColumn;
