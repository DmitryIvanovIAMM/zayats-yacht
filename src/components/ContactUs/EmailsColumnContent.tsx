import React from 'react';
import { secondary } from '../colors';

const emailStyle = {
  textDecoration: 'none',
  color: secondary.dark
};

export default function EmailsColumnContent() {
  return (
    <div style={{ fontSize: '16px', lineHeight: '150%' }}>
      <br />
      <a href="mailto:CustomerService@zayats-yacht.com" style={emailStyle}>
        CustomerService@zayats-yacht.com
      </a>
    </div>
  );
}
