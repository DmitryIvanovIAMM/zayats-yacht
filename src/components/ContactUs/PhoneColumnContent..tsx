import React from 'react';
import { secondary } from '../colors';

export default function PhoneColumnContent() {
  return (
    <div style={{ fontSize: '16px', lineHeight: '150%' }}>
      <br />
      Tel:&nbsp;
      <a href="tel:+15555555555" style={{ textDecoration: 'none', color: secondary.dark }}>
        +1 555-555-5555
      </a>
      <br />
    </div>
  );
}
