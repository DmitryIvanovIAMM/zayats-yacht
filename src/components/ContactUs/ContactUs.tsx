import SectionTitle from '@/components/SectionTitle/SectionTitle';
import ContactUsColumn from '@/components/ContactUs/ContactUsColumn';
import Box from '@mui/material/Box';
import AddressColumnContent from '@/components/ContactUs/AddressColumnContent';
import PhoneColumnContent from '@/components/ContactUs/PhoneColumnContent.';
import EmailsColumnContent from '@/components/ContactUs/EmailsColumnContent';
import LocationOnIcon from '@mui/icons-material/LocationOn';
import CallIcon from '@mui/icons-material/Call';
import EmailIcon from '@mui/icons-material/Email';

export const sectionContainerSx = {
  display: 'flex',
  justifyContent: 'center',
  justifyItems: 'center',
  alignItems: 'center',
  flexDirection: 'column',
  marginLeft: '20px',
  marginRight: '20px',
  marginTop: '40px',
  marginBottom: '20px',
  color: 'secondary.dark'
};
const contactUsSectionSx = {
  ...sectionContainerSx,
  margin: '0px',
  backgroundColor: '#F7F8F9',
  padding: '10px',
  width: `100%`
};
const contactUsDataSx = {
  display: 'flex',
  marginTop: {
    xs: '0px',
    sm: '30px'
  },
  justifyContent: 'space-around',
  alignItems: {
    xs: 'left',
    md: 'start'
  },
  flexDirection: {
    xs: 'column',
    sm: 'row'
  },
  color: 'secondary.dark'
};

const ContactUs = () => {
  return (
    <Box id="contact-us-section" sx={contactUsSectionSx}>
      <SectionTitle title="Contact Us" />
      <Box sx={contactUsDataSx}>
        <ContactUsColumn title="Address" icon={LocationOnIcon}>
          <AddressColumnContent />
        </ContactUsColumn>
        <ContactUsColumn title="Phone" icon={CallIcon}>
          <PhoneColumnContent />
        </ContactUsColumn>
        <ContactUsColumn title="Email" icon={EmailIcon}>
          <EmailsColumnContent />
        </ContactUsColumn>
      </Box>
    </Box>
  );
};

export default ContactUs;
