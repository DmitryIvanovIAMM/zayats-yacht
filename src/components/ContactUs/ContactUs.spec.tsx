import { render } from '@testing-library/react';
import ContactUs from '@/components/ContactUs/ContactUs';

const setup = async () => {
  const container = render(<ContactUs />);
  return {
    container
  };
};

describe('ContactUs component', () => {
  it('should render correctly', async () => {
    const { container } = await setup();

    expect(container).toMatchSnapshot();
  });

  it('should show contact information', async () => {
    const { container } = await setup();

    expect(container.getByText('Address')).toBeInTheDocument();
    expect(container.container).toHaveTextContent('Zayats Yacht Transport');
    expect(container.container).toHaveTextContent('55555 Orange Drive, Suite 555');
    expect(container.container).toHaveTextContent('Fort Lauderdale, FL, +55555, USA');

    expect(container.getByText('Phone')).toBeInTheDocument();
    expect(container.container).toHaveTextContent('+1 555-555-5555');

    expect(container.getByText('Email')).toBeInTheDocument();
    expect(container.container).toHaveTextContent('CustomerService@zayats-yacht.com');
  });
});
