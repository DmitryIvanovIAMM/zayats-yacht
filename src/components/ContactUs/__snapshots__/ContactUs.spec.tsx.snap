// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContactUs component should render correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiBox-root css-lyi2jj"
        id="contact-us-section"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-10dtpk3-MuiTypography-root"
          style="--Typography-textAlign: center;"
        >
          Contact Us
        </h4>
        <div
          class="MuiBox-root css-xt539u"
        >
          <div
            class="MuiBox-root css-1oj1pdk"
          >
            <div
              style="background: rgb(255, 255, 255); box-shadow: 0px 2px 20px rgba(51, 101, 167, 0.18); border-radius: 50%; min-width: 67px; max-width: 67px; height: 67px; display: flex; align-items: center; justify-content: center; margin-right: 20px;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-uks7c6-MuiSvgIcon-root"
                data-testid="LocationOnIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5"
                />
              </svg>
            </div>
            <div
              style="align-items: start;"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-1fs9uky-MuiTypography-root"
              >
                <br />
                Address
              </h6>
              <div
                style="font-size: 16px; line-height: 150%;"
              >
                <br />
                Allied Yacht Transport, LLC
                <br />
                12555 Orange Drive, Suite 228
                <br />
                Fort Lauderdale, FL, 33330, USA
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-1oj1pdk"
          >
            <div
              style="background: rgb(255, 255, 255); box-shadow: 0px 2px 20px rgba(51, 101, 167, 0.18); border-radius: 50%; min-width: 67px; max-width: 67px; height: 67px; display: flex; align-items: center; justify-content: center; margin-right: 20px;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-uks7c6-MuiSvgIcon-root"
                data-testid="CallIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20.01 15.38c-1.23 0-2.42-.2-3.53-.56-.35-.12-.74-.03-1.01.24l-1.57 1.97c-2.83-1.35-5.48-3.9-6.89-6.83l1.95-1.66c.27-.28.35-.67.24-1.02-.37-1.11-.56-2.3-.56-3.53 0-.54-.45-.99-.99-.99H4.19C3.65 3 3 3.24 3 3.99 3 13.28 10.73 21 20.01 21c.71 0 .99-.63.99-1.18v-3.45c0-.54-.45-.99-.99-.99"
                />
              </svg>
            </div>
            <div
              style="align-items: start;"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-1fs9uky-MuiTypography-root"
              >
                <br />
                Phone
              </h6>
              <div
                style="font-size: 16px; line-height: 150%;"
              >
                <br />
                Tel: 
                <a
                  href="tel:+19543944640"
                  style="text-decoration: none; color: rgb(0, 102, 102);"
                >
                  +1 954-394-4640
                </a>
                <br />
              </div>
            </div>
          </div>
          <div
            class="MuiBox-root css-1oj1pdk"
          >
            <div
              style="background: rgb(255, 255, 255); box-shadow: 0px 2px 20px rgba(51, 101, 167, 0.18); border-radius: 50%; min-width: 67px; max-width: 67px; height: 67px; display: flex; align-items: center; justify-content: center; margin-right: 20px;"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-uks7c6-MuiSvgIcon-root"
                data-testid="EmailIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 4-8 5-8-5V6l8 5 8-5z"
                />
              </svg>
            </div>
            <div
              style="align-items: start;"
            >
              <h6
                class="MuiTypography-root MuiTypography-h6 css-1fs9uky-MuiTypography-root"
              >
                <br />
                Email
              </h6>
              <div
                style="font-size: 16px; line-height: 150%;"
              >
                <br />
                <a
                  href="mailto:sales@allied-yacht.com"
                  style="text-decoration: none; color: rgb(0, 102, 102);"
                >
                  sales@allied-yacht.com
                </a>
                <br />
                <a
                  href="mailto:operations@allied-yacht.com"
                  style="text-decoration: none; color: rgb(0, 102, 102);"
                >
                  operations@allied-yacht.com
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiBox-root css-lyi2jj"
      id="contact-us-section"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter css-10dtpk3-MuiTypography-root"
        style="--Typography-textAlign: center;"
      >
        Contact Us
      </h4>
      <div
        class="MuiBox-root css-xt539u"
      >
        <div
          class="MuiBox-root css-1oj1pdk"
        >
          <div
            style="background: rgb(255, 255, 255); box-shadow: 0px 2px 20px rgba(51, 101, 167, 0.18); border-radius: 50%; min-width: 67px; max-width: 67px; height: 67px; display: flex; align-items: center; justify-content: center; margin-right: 20px;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-uks7c6-MuiSvgIcon-root"
              data-testid="LocationOnIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C8.13 2 5 5.13 5 9c0 5.25 7 13 7 13s7-7.75 7-13c0-3.87-3.13-7-7-7m0 9.5c-1.38 0-2.5-1.12-2.5-2.5s1.12-2.5 2.5-2.5 2.5 1.12 2.5 2.5-1.12 2.5-2.5 2.5"
              />
            </svg>
          </div>
          <div
            style="align-items: start;"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-1fs9uky-MuiTypography-root"
            >
              <br />
              Address
            </h6>
            <div
              style="font-size: 16px; line-height: 150%;"
            >
              <br />
              Allied Yacht Transport, LLC
              <br />
              12555 Orange Drive, Suite 228
              <br />
              Fort Lauderdale, FL, 33330, USA
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1oj1pdk"
        >
          <div
            style="background: rgb(255, 255, 255); box-shadow: 0px 2px 20px rgba(51, 101, 167, 0.18); border-radius: 50%; min-width: 67px; max-width: 67px; height: 67px; display: flex; align-items: center; justify-content: center; margin-right: 20px;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-uks7c6-MuiSvgIcon-root"
              data-testid="CallIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20.01 15.38c-1.23 0-2.42-.2-3.53-.56-.35-.12-.74-.03-1.01.24l-1.57 1.97c-2.83-1.35-5.48-3.9-6.89-6.83l1.95-1.66c.27-.28.35-.67.24-1.02-.37-1.11-.56-2.3-.56-3.53 0-.54-.45-.99-.99-.99H4.19C3.65 3 3 3.24 3 3.99 3 13.28 10.73 21 20.01 21c.71 0 .99-.63.99-1.18v-3.45c0-.54-.45-.99-.99-.99"
              />
            </svg>
          </div>
          <div
            style="align-items: start;"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-1fs9uky-MuiTypography-root"
            >
              <br />
              Phone
            </h6>
            <div
              style="font-size: 16px; line-height: 150%;"
            >
              <br />
              Tel: 
              <a
                href="tel:+19543944640"
                style="text-decoration: none; color: rgb(0, 102, 102);"
              >
                +1 954-394-4640
              </a>
              <br />
            </div>
          </div>
        </div>
        <div
          class="MuiBox-root css-1oj1pdk"
        >
          <div
            style="background: rgb(255, 255, 255); box-shadow: 0px 2px 20px rgba(51, 101, 167, 0.18); border-radius: 50%; min-width: 67px; max-width: 67px; height: 67px; display: flex; align-items: center; justify-content: center; margin-right: 20px;"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-uks7c6-MuiSvgIcon-root"
              data-testid="EmailIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2m0 4-8 5-8-5V6l8 5 8-5z"
              />
            </svg>
          </div>
          <div
            style="align-items: start;"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 css-1fs9uky-MuiTypography-root"
            >
              <br />
              Email
            </h6>
            <div
              style="font-size: 16px; line-height: 150%;"
            >
              <br />
              <a
                href="mailto:sales@allied-yacht.com"
                style="text-decoration: none; color: rgb(0, 102, 102);"
              >
                sales@allied-yacht.com
              </a>
              <br />
              <a
                href="mailto:operations@allied-yacht.com"
                style="text-decoration: none; color: rgb(0, 102, 102);"
              >
                operations@allied-yacht.com
              </a>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
